src/Validate/PriceConstraint.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Drupal\tide_event\Validate;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+
+/**
+ * Form API callback. Validate element value.
+ */
+class PriceConstraint {
+
+  /**
+   * Validates given element.
+   *
+   * @param array $element
+   *   The form element to process.
+   * @param \Drupal\Core\Form\FormStateInterface $formState
+   *   The form state.
+   * @param array $form
+   *   The complete form structure.
+   */
+  public static function validate(array &$element, FormStateInterface $formState, array &$form) {
+    $error = FALSE;
+    $webformKey = $element['#webform_key'];
+    $price_from_value = $formState->getValue('price_from');
+    $price_to_value = $formState->getValue($webformKey);
+
+    // Skip empty unique fields or arrays (aka #multiple).
+    if ($price_to_value === '' || is_array($price_to_value)) {
+      return;
+    }
+
+    if ($price_to_value < $price_from_value) {
+      $error = TRUE;
+    }
+
+    if ($error) {
+      if (isset($element['#title'])) {
+        $tArgs = [
+          '%name' => empty($element['#title']) ? $element['#parents'][0] : $element['#title'],
+          '%value' => $price_to_value,
+        ];
+        $formState->setError(
+          $element,
+          new TranslatableMarkup('The %name value "%value" should be greater or equal to "Price From" value. Please use a different amount.', $tArgs)
```

Comment:
@enzolutions both `t()` and `TranslatableMarkup` should not be used directly. You should instead use dependency injection to call the translation service. See https://github.com/sonnykt/module_permissions/blob/8.x-1.x/src/UninstallValidator.php

Secondly, why do we need a new class just for form element validation? Don't just follow the [webform article](https://www.drupal.org/docs/8/modules/webform/webform-cookbook/how-to-add-custom-validation-to-a-webform-element) as it's overkilled. You only need a new class if you want to define a [constraint validator](https://www.drupal.org/docs/8/api/entity-validation-api/providing-a-custom-validation-constraint) and your constraint can be used elsewhere. And standard validate callback is enough and you don't have to deal with dependency injection.

Reply:
OK I will use the Trait.

About the webform I just follow the official documentation for Webform https://www.drupal.org/docs/8/modules/webform/webform-cookbook/how-to-add-custom-validation-to-a-webform-element

Reply:
@sonnykt Just picking this up now that Enzo has left. It doesn't look like this has progressed with the changes you requested. Are there changes that didn't get pushed to this branch that I don't know about? I agree with injecting `TranslatableMarkup` (good pickup!) but could be convinced either way re: procedural versus OO. What is the best way to get this ticket completed?